r1cs/core/src/constraint_system.rs

```diff
@@ -1,9 +1,29 @@
+use std::collections::HashMap;
 use algebra::Field;
 use radix_trie::Trie;
 use std::marker::PhantomData;
 
 use crate::{Index, LinearCombination, SynthesisError, Variable};
 
+/// Instrument a code block $body to count the constraints enforced inside the block in the
+/// counter of constraint system $cs identified by $label
+//ToDo: reasoning on improving this macro to avoid the need to enclose the whole code block inside
+// a closure, maybe relying on existing crates or by writing a procedural macro
```

Comment:
I would say also find a way to do it with the already existing namespaces (we are introducing other labels but we already have the namespaces, sounds like a duplication. If one designs the namespaces carefully, maybe like arkworks did, then we can do better)

Reply:
Not sure it will be robust enough to rely on namespaces (unless we define a precise set of unambiguous rules to construct them and we are willing to modify all existing namespaces to comply with such rules); btw, I add this possibility to the ToDo to ensure we will take it into account in future discussions.